test: Ignore UTF-8 errors in assert_debug_log

maflcko opened the pull request:
Fix two bugs, see commit messages.

Commit:
test: Ignore UTF-8 errors in assert_debug_log

read() fails in text mode when the unicode hasn't been fully written
yet. Fixes: "wallet_importdescriptors.py: can't decode bytes in position
228861-228863: unexpected end of data"
(https://github.com/bitcoin/bitcoin/issues/28030)

Commit:
lint: Ignore check_fileopens failure on **kwargs

This fixes a bug in the linter:

"""
Python's open(...) seems to be used to open text files without explicitly specifying encoding='utf8':
test/functional/test_framework/test_node.py:        with open(self.debug_log_path, **kwargs) as dl:
"""

www222fff reviewed (commented):
Is this fix applied to both for python2 and python3?

Commit:
test: Fix debug_log_size helper

debug_log_bytes returned "an opaque number when in text mode"
(https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects),
not the number of bytes.

Fix this by using binary mode or text mode (with the same encoding)
consistently when opening the file for ftell() and read().

theStack reviewed (approved):
utACK fa3d72960bc86319aa8b838e3df4e833f845c71f

Changes look correct to me and should fix #28030.
(Background information for other reviewers: `wait_for_debug_log` was changed to open the debug file in binary mode in PR #25294)
